TrackletAlgorithm/MatchProcessor.h

```diff
@@ -590,50 +1263,46 @@ void MatchProcessor(BXType bx,
   static ap_uint<2*MEBinsBits> zbinLUT[128];
 #pragma HLS ARRAY_PARTITION variable=zbinLUT complete dim=0
   zbinLUTinit(zbinLUT, zbins_adjust_PSseed, zbins_adjust_2Sseed);
+  constexpr int nRbinBits = VMProjection<VMPTYPE>::kVMProjFineZSize + VMProjectionBase<VMPTYPE>::kVMProjZBinSize;
+  constexpr int nRbin = 1<<TrackletProjection<PROJTYPE>::kTProjRZSize;
+  static ap_uint<nRbinBits> rbinLUT[256];//1<<TrackletProjection<PROJTYPE>::kTProjRZSize];
+#pragma HLS ARRAY_PARTITION variable=rbinLUT complete dim=0
+  readRbin_LUT<LAYER,nRbinBits,256>(rbinLUT);
+
+  const auto LUT_matchcut_rbend_width = 5;
+  const auto LUT_matchcut_rbend_depth = 4096;
+  ap_uint<LUT_matchcut_rbend_width> LUT_matchcut_rbend[LUT_matchcut_rbend_depth];
+  readTable_rbend<MC::RBEND,LAYER,LUT_matchcut_rbend_width,LUT_matchcut_rbend_depth>(LUT_matchcut_rbend);
+  // Initialize MC delta phi cut variables
+  ap_uint<MC::LUT_matchcut_z_width> best_delta_z;
+  ap_uint<MC::LUT_matchcut_phi_width> best_delta_phi;
+  ap_uint<MC::LUT_matchcut_rphi_width> best_delta_rphi;
+  ap_uint<MC::LUT_matchcut_r_width> best_delta_r;
+  ap_uint<1> isMatch = 0;
+  ap_uint<1> hasMatch = 0;
+  ap_uint<2> bestiMEU = 0;
 
-  bool validproj__ = false;
-  ProjectionRouterBuffer<BARREL, APTYPE> projbuffer__;
-
-
-  ap_uint<8> vmstubsmask[8];
-#pragma HLS array_partition variable=vmstubsmask complete dim=1
-
-  entriesloop:for(unsigned int i=0;i<8;i++) {
-#pragma HLS unroll
-    vmstubsmask[i]=instubdata.getBinMask8(bx,i);
-  }
 // constants used in reading VMSME memories
   constexpr int NUM_PHI_BINS = 1 << kNbitsphibin;
   constexpr int NUM_RZ_BINS = 1 << kNbitsrzbin;
   constexpr int PAGE_LENGTH_CM = 1024;
   constexpr int BIN_ADDR_WIDTH = 4;
-  //Can't figure out how to do this calculation at compile
-  //constexpr int BIN_ADDR_WIDTH = ceil(log2(PAGE_LENGTH_CM/(NUM_PHI_BINS*NUM_RZ_BINS)));
-
-  /*
-  ap_uint<4> nvmstubs[8][8]; 
-#pragma HLS ARRAY_PARTITION variable=nvmstubs complete dim=0
-  
- nstubsloop: for (unsigned int izbin=0;izbin<8;izbin++) {
-#pragma HLS unroll      
-    (nvmstubs[izbin][7],nvmstubs[izbin][6],nvmstubs[izbin][5],nvmstubs[izbin][4],
-     nvmstubs[izbin][3],nvmstubs[izbin][2],nvmstubs[izbin][1],nvmstubs[izbin][0]) = instubdata.getEntries8(bx, izbin);
-  }
-  */
  PROC_LOOP: for (ap_uint<kNBits_MemAddr> istep = 0; istep < kMaxProc - kMaxProcOffset(module::MP); istep++) {
 #pragma HLS PIPELINE II=1 rewind
+    if (hasMatch)
+      matchengine[bestiMEU].advance();
```

Comment:
What is the reason to move this code here?

Reply:
This was a timing fix I had implemented. I haven't tested moving it back after @jasonfan393's PR was merged.